### PR TITLE
Adding vuid 02784 02785, Validate SubgroupSizeControlFeaturesEXT 

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Check if SubgroupSizeControlFeaturesEXT are enabled when used

### CODE REVIEW

File: layers/shader_validation.cpp [hunk 1]
```diff
@@ -2098,6 +2098,23 @@ bool CoreChecks::ValidatePipelineShaderStage(VkPipelineShaderStageCreateInfo con
     if (pStage->stage == VK_SHADER_STAGE_COMPUTE_BIT) {
         skip |= ValidateComputeWorkGroupSizes(module, entrypoint);
     }
+
+    if ((pStage->flags & VK_PIPELINE_SHADER_STAGE_CREATE_ALLOW_VARYING_SUBGROUP_SIZE_BIT_EXT) != 0 &&
```
Comment (sfricke-samsung) [inline layers/shader_validation.cpp:2102]: this function is already kinda large, might be slightly nicer to make this in its own function and go

```cpp
    if (IsExtEnabled(device_extensions.vk_qcom_render_pass_shader_resolve)) {
        skip |= ValidateShaderResolveQCOM(module, pStage, pipeline);
    }
    if (IsExtEnabled(device_extensions.vk_ext_subgroup_size_control)) {
        skip |= ValidateShaderSubgroupSizeControl(pStage);
    }
```

File: tests/vklayertests_pipeline_shader.cpp [hunk 2]
```diff
@@ -11749,7 +11749,12 @@ TEST_F(VkLayerTest, PipelineSubgroupSizeControl) {
         return;
     }
     m_device_extension_names.push_back(VK_EXT_SUBGROUP_SIZE_CONTROL_EXTENSION_NAME);
-    ASSERT_NO_FATAL_FAILURE(InitState());
+
+    VkPhysicalDeviceSubgroupSizeControlFeaturesEXT sscf = LvlInitStruct<VkPhysicalDeviceSubgroupSizeControlFeaturesEXT>();
+    sscf.subgroupSizeControl = VK_TRUE;
+
+    VkPhysicalDeviceFeatures2 pd_features2 = LvlInitStruct<VkPhysicalDeviceFeatures2>(&sscf);
+    ASSERT_NO_FATAL_FAILURE(InitState(nullptr, &pd_features2));
```
Comment (jeremyg-lunarg) [inline tests/vklayertests_pipeline_shader.cpp:11757]: Was this test broken before?

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: tests/vklayertests_pipeline_shader.cpp (hunk 2 above)
Comment (ziga-lunarg) [reply tests/vklayertests_pipeline_shader.cpp:11757]: Yes, the used feature was not enabled, caught by this added vuid.

**Pull request merged**